lib/dialects/abstract/query-generator.js

```diff
@@ -249,8 +249,10 @@ const QueryGenerator = {
     }
 
     if (this._dialect.supports.updateOnDuplicate && options.updateOnDuplicate) {
-      onDuplicateKeyUpdate += ' ON DUPLICATE KEY UPDATE ' + options.updateOnDuplicate.map(attr => {
-        const field = rawAttributes && rawAttributes[attr] && rawAttributes[attr].field || attr;
```

Comment:
This should be enough, what do you observe here, `rawAttributes` empty?

Reply:
Can you work on test case first? `rawAttributes` must not be undefined, we can change this method parameters OR pass `rawAttributes` from methods which don't pass it OR use `options.model` instead.

Let me know if you get stuck, I will handle this fix myself then :)

Reply:
Hi! Yeah, i was working on test case but i have been travelling some days and i need to stop. I will push test in this week ;)

Reply:
I'm executing integration tests with docker but your referenced test for updateOnDuplicate option doesn't appear in the launched collection. Any idea? Am i doing something wrong?
Also, when i launch integration tests, 255 passing and 201 failing...